### PR TITLE
Add the `batch_send` endpoint to generic workers

### PR DESCRIPTION
**Pull request by Fizzadar**
I believe this should be safe, just as general room event sending is safe on worker instances. Another thing that can be pulled off main.

Signed off by Nick @ Beeper.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: changelog.d/12868.misc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Enable the `batch_send` endpoint on synapse worker processes. Contributed by Nick @ Beeper.
+
```
Comment (erikjohnston) [inline changelog.d/12868.misc:2]:
```suggestion
Enable the `batch_send` endpoint on synapse worker processes. Contributed by Nick @ Beeper.
```

Comment (erikjohnston) [inline changelog.d/12868.misc:2]: Can you remove the trailing newline please? Annoyingly GH is refusing to let me make a suggested change to do that :(

**Review by erikjohnston — COMMENTED** [review]
Can you add this endpoint to the complement worker config please: https://github.com/matrix-org/synapse/blob/774ac4930dbb0e6f2f6dad4b9eb4630154e1e161/docker/configure_workers_and_start.py#L160 ? We don't currently run it in CI, but when we do it'll make sure we don't accidentally break it

**Review by erikjohnston — APPROVED** [review]
Woop :rocket:

File: docs/workers.md [hunk 2]
```diff
@@ -206,6 +206,7 @@ information.
     ^/_matrix/client/(api/v1|r0|v3|unstable)/rooms/.*/members$
     ^/_matrix/client/(api/v1|r0|v3|unstable)/rooms/.*/state$
     ^/_matrix/client/(v1|unstable/org.matrix.msc2946)/rooms/.*/hierarchy$
+    ^/_matrix/client/(v1|unstable/org.matrix.msc2716)/rooms/.*/batch_send$
```
Comment (clokep) [inline docs/workers.md:209]: This (and other references) are incorrect, it is not available under a `v1` endpoint yet, only the unstable version.

**Pull request merged**